woocommerce/google-listings-and-ads
# Modify `path` in URL to make additional pages work with WooCommerce Navigation

### Changes proposed in this Pull Request:

Closes #665 .

The PR changes the `path` for the following five pages so that it works nicely with WC Navigation sidebar.

- Programs Report
- Products Report
- Edit Free Listings
- Add another paid campaign
- Edit paid campaign

For the report pages, the `path` is now `/google/reports`, with another query string key `reportKey` with value `programs` or `products`. 

For the add and edit pages, the `path` is now `/google/dashboard`, with another query string key `subpath` with one of the following values: 

- `/free-listings/edit`
- `/campaigns/edit`
- `/campaigns/create`

#### Technical notes:

##### PHP files

Because of the path move, some PHP files are not needed anymore, I have removed them in this PR too.

##### `FullContainer` component

There was a quite a bit of challenge to make `FullContainer` work with both classic navigation and the new WC navigation. It is used in the add and edit pages. I made some code changes to `FullContainer` by referring to the code in WC Admin:

https://github.com/woocommerce/woocommerce-admin/blob/95c487247416ab34eb8e492b984e2b068618e0d3/client/header/index.js#L92-L118

WC Admin applies `margin-top` to `#wpbody` after a 200ms timeout. To counter that, we do a similar timeout approach and apply a negative `margin-top` to the child `#wpbody-content`.

### Screenshots:

Demo video with new WC Navigation: (https://d.pr/v/vlw0Ok)

https://user-images.githubusercontent.com/417342/120803644-5a7ebc00-c576-11eb-8b53-8c9cab2c101a.mov

Demo video with classic navigation: (https://d.pr/v/JQX0vQ)

https://user-images.githubusercontent.com/417342/120803658-5c487f80-c576-11eb-803a-44518cdeb95b.mov

### Detailed test instructions:

1. Go to https://gla1.test/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features and turn on WC Navigation.
2. Visit all the pages in GLA (especially the five pages mentioned above). Make sure that when you are viewing the pages, the WC Navigation side bar is showing the GLA level menu and not the WC root level menu.
3. When you are viewing the five pages mentioned above, reload the page and make sure the page is displaying correctly. (This is specifically to test the `FullContainer` changes.)
4. Go to https://gla1.test/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features and turn **off** WC Navigation.
5. Repeat step 2 and 3 above.

### Changelog Note:

Modify `path` in URL to make additional pages work with WooCommerce Navigation
